File: C/credit_card/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number, size, container, container_two, digit, digit_two, num;
    long count = 0, sum = 0, sum_two = 0, digit_x = 0, digit_y = 0;
    int start_digit, start_digit_two;

    //Get the number from user
    do
    {
        number = get_long("Number: ");
    }
    while (number < 1);

    //Sum of the number’s second-to-last digit
    container = number;
    for (long i = 100 ; i <= number * 10 ; i *= 100)
    {
        digit = (container % 100) / 10;
        container = number / i;
        if ((digit * 2) >= 10)
        {
            digit_x += (digit * 2) / 10;
            digit_y += (digit * 2) % 10;
        }
        else
        {
            sum += (digit * 2);
        }
    }
    long db_sum = sum + digit_x + digit_y;

    //Sum digits that weren’t multiplied by 2
    container_two = number;
    for (long j = 100 ; j <= number * 10 ; j *= 100)
    {
        digit_two = (container_two % 10);
        container_two = number / j;
        sum_two += digit_two;
        if (container_two < 10)
        {
            sum_two += container_two;
            break;
        }
        else
        {
            continue;
        }
    }

    long total = db_sum + sum_two;

    //Count the size of the number
    size = number;
    while (size != 0)
    {
        size /= 10;
        count ++;
        if (size < 100 && size > 10)
        {
            start_digit = size;
        }
        if (size < 10 && size > 0)
        {
            start_digit_two = size;
        }
    }

    //Validating credit card numbers
    if (total % 10 == 0)
    {
        if (count == 15 && start_digit == 34 || start_digit == 37)
        {
            printf("AMEX\n");
        }
        else if (count == 16 && start_digit == 51 || start_digit == 52 || start_digit == 53 || start_digit == 54 || start_digit == 55)
        {
            printf("MASTERCARD\n");
        }
        else if (count == 13 || count == 16 && start_digit_two == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }

    }
    else
    {
        printf("INVALID\n");
    }
}

File: Python/credit/credit.py
import cs50


# main function
def main():
    # get the number from user
    number = cs50.get_string('Number: ')

    # append the each number to the list and reverse the number list
    re_number_list = []
    for i in number:
        re_number_list.append(i)
    re_number_list.reverse()

    sum_x = 0
    sum_y = 0
    number_product = 0
    tmp_product = []
    final_product = []
    for i in range(1, len(re_number_list), 2):
        number_product = int(re_number_list[i]) * 2
        tmp_product.append(number_product)

    for k in tmp_product:
        if k > 9:
            num_tenth = int(k / 10)
            num_digit = k % 10
            final_product.append(num_tenth)
            final_product.append(num_digit)
        else:
            final_product.append(k)

    # sum of the number’s second-to-last digit
    for n in final_product:
        sum_x += n

    # sum digits that weren’t multiplied by 2
    for j in range(0, len(re_number_list), 2):
        sum_y += int(re_number_list[j])

    number_list = []
    for i in number:
        number_list.append(i)

    start_digit_a = int(number_list[0])
    start_digit_b = int(number_list[0] + number_list[1])

    # validating credit card numbers
    if (sum_x + sum_y) % 10 == 0:
        if start_digit_a == 4 and (len(number) == 13 or len(number) == 16):
            print('VISA')

        elif len(number) == 15 and (start_digit_b == 37 or start_digit_b == 34):
            print('AMEX')

        elif start_digit_a != 4 and len(number) == 16:
            if start_digit_b >= 51 and start_digit_b <= 55:
                print('MASTERCARD')
            else:
                print('INVALID')
        else:
            print('INVALID')

    else:
        print('INVALID')


if __name__ == "__main__":
    main()
